### DISCUSSION TITLE
Problems with 4.0 release - extension is very slow to load

### DISCUSSION THREAD

**Discussion by jsc85** [gandf/slick-rss#58] [General]
I am using Chrome 128.0.6613.114 on Windows 11. When I open up Slick RSS, I see this gray screen for a very long time. Eventually the feeds load, but very, very slowly. Similarly, if I switch to another part of the extension, like manage, it takes a very long time to load. This started with the 4.0 release. Before then, it was mainly working. I tried reverting my extension directory, and the extension worked for a few min before it loaded 4.0 again, wherein the problems restarted. Below is a screenshot of what I am seeing.

![Screenshot 2024-09-01 193531](https://github.com/user-attachments/assets/16327890-e5f4-43ae-b618-759e0a713bdd)

**Comment by gandf** [comment]
Hi,

Try to disable extension and reenable.
If issue persiste, push "Refresh all" link (page up left).

Can you please specify what CPU you have and total amount of RAM?

If you go to manage, you have your feeds ?
If you go to option, do you have your option ?
Can you post a screeshot of your options ?

Can you go to "My extensions" and verify if error are present ? Can you post error ?
![image](https://github.com/user-attachments/assets/813cb07b-62e3-4860-9a1e-7076a9d7fcdb)

Have in advance and I'm sorry for issue.

**Comment by jsc85** [comment]
I've reinstalled/disabled a few times and it doesn't help. The only relief I get is going into developer mode and switching back to 3.17, as otherwise it auto updates to 4.0.0 and the issues start again.

I have a AMD Ryzen 7 5700G with 32 GB of RAM.

I have my feeds in manage, and option works. I don't have any errors when outside developer mode.
![image](https://github.com/user-attachments/assets/dc6e946d-28f7-4633-9bf9-ba70c58ef6d3)

**Comment by gandf** [comment]
I requested a rollback. Version 4.0.1 is the new name for 3.17.

There is often a difference in operation between dev mode and prod. It also depends on the browser.
I develop on chrome and use on vivaldi.
I saw that there is another person who complains about having an error message indicating that a file is missing.
My PC is close to yours (Ryzen 7 7700 with 64GB). I will test on my latop with more browser.

The tests I did showed that there was a small blockage during the data conversion and I solved it. But I see that once in prod, there seems to be a difference in operation and I see that sometimes he freezes.

I suspect the offscreen document or workers have a much shorter lifespan in production.

**Comment by gandf** [comment]
I have a Ilast question.
Can you option a page (viewer or option)

Right click on page and choose "inspect" or F12.
![image](https://github.com/user-attachments/assets/0446a80c-6f2f-40e0-afd7-96f1301d78a9)

In new opened page, choose "Application" , go to IndexedDB=> SlickRssByUsers=>keyvaluepairs and take a screenshoot.
![image](https://github.com/user-attachments/assets/6ae5661b-661b-4996-8119-af478fef013b)

It's to see if convertion has worker or not.
I converted the data but kept the original data so I could roll back if something went wrong.

This will give me a hint on which step is not working.

Thanks in advance.

**Comment by jsc85** [comment]
I was away for a few days. I reloaded 4.0.0 to answer the question, and it appears to be working now. 
![image](https://github.com/user-attachments/assets/d92c405c-88cf-4eab-bd3f-97f9bdead794)

**Comment by gandf** [comment]
Thanks.

Tables whose name begins with “Table” are created by the new extension.
All others are the old.

This means that tables have been converted and that offscreen document and web worker system to run SQL is working properly.

I thought it was a right issue and that the whole system didn't work once it was in production.

Now I think it's a problem of shelf life. This has to close and the time to search for everything has to be long.
Otherwise, there's a value somewhere that makes the system do nothing.

For my part, I've noticed that the system is very responsive when I use the sources directly, but that in prod I sometimes have to double-click on the menu for it to open, or I have to click on another menu to finally open it.
I don't have this problem with sources. I'll do some more tests this week to fix this case and hope it's the same origin as initial block.
